frontend/subscriptions/index.js

```diff
@@ -0,0 +1,49 @@
+import {
+  main$,
+  appWillStart$,
+  navigate$,
+  historyPush,
+  getCurrentRoute,
+  INDEX_PATH,
+} from '@shopgate/engage/core';
+import { logger } from '@shopgate/pwa-core/helpers';
+import { getSelection } from '../selectors';
+import { getStartpage, updateSelection } from '../actions';
+import { SET_SELECTION } from '../constants';
+
+export default (subscribe) => {
+  const indexDidEnter$ = navigate$.filter(({ action }) => action.params.pathname === INDEX_PATH);
+  const setSwitchSelection$ = main$.filter(({ action }) => action.type === SET_SELECTION);
+
+  subscribe(indexDidEnter$, ({ dispatch, getState }) => {
+    const state = getState();
+    const selection = getSelection(state);
+    const route = getCurrentRoute(state);
+
+    // home button redirects to selection
+    if (selection.path !== route.pathname) {
+      dispatch(historyPush({
+        pathname: selection.path,
+      }));
+    }
+  });
+
+  subscribe(setSwitchSelection$, ({ dispatch, getState }) => {
+    const selection = getSelection(getState());
+
+    dispatch(historyPush({
+      pathname: selection.path,
+    }));
+  });
+
+  subscribe(appWillStart$, async ({ dispatch, getState }) => {
+    try {
+      await dispatch(getStartpage());
+      const selection = getSelection(getState());
+
+      dispatch(updateSelection(selection));
```

Comment:
`updateSelection` right now eventually emits the `setSwitchSelection$` and users are redirected to the previous selection. that causes that it's not possible to open the PWA with any page other than the current selection.

i'm still not sure if this causes an issue in real life - especially when the app is opened via a push message and the page from the push is opened before the response from `updateSelection` came in.

maybe we just should give the current approach a try since right now i also don't have a much better idea.

Reply:
we have not yet been able to test it. this will be possible with the live release (in 3-4 weeks).